### PR TITLE
pin supported metadata schema versions in Redshift loader

### LINKED ISSUE #341: Update subscription to DSS to only receive data adhering to compatible schemas
This is required in order to receive data that adheres to a schema versions with which matrix service is compatible.

### PR DESCRIPTION
**Pull request by calvinnhieu**
Closes #341 

In accordance with Phase 1.4 of the [Querying DSS by Metadata Schema Version(s) RFC](https://github.com/HumanCellAtlas/dcp-community/pull/90/files?short_path=41a4263#diff-41a4263df7296b16ac19395c05f3ebca)

### CODE REVIEW

File: scripts/redshift/loader.py [hunk 1]
```diff
@@ -2,11 +2,59 @@
...
+    },
+    MetadataSchemaName.LIBRARY_PREPARATION_PROTOCOL: {
+        'major': 6,
+        'minor': 1
+    },
+    MetadataSchemaName.ANALYSIS_PROTOCOL: {
+        'major': 9,
+        'minor': 0
+    },
+    MetadataSchemaName.SPECIMEN_FROM_ORGANISM: {
```
Comment (mckinsel) [inline scripts/redshift/loader.py]: We also have minimum supported versions. For example, for `specimen_from_organism`, it's 9.0.0.

File: scripts/redshift/loader.py [hunk 2]
```diff
@@ -96,6 +145,57 @@ def load(args):
...
+    :param schema_name: MetadataSchemaName to generate clause for
+    :return: dict ES query clause
+    """
+    return {
+        "bool": {
+            "should": [
+                {
+                    "bool": {
+                        "must_not": {
+                            "exists": {
```
Comment (mckinsel) [inline scripts/redshift/loader.py]: Maybe we can add some comments around these? To me (with not much ES knowledge), this looks like we don't want bundles that have the `schema_major_version` field. Does that mean we would accept bundles without the schema version annotated this way?

Comment (calvinnhieu) [reply scripts/redshift/loader.py]: Sure, will add comments. 

Yes, specifically, this `must_not` `exists` is in an "OR" (`should`) clause that will match bundles that either 1) have `schema_*_version` fields defined and is within our supported version range or 2) do not have `schema_*_version` fields defined. 

The importance of (2) is to continue to match pre-schema-version bundles before existing bundles are backfilled with the new schema fields. This shouldn't be an issue as we currently support these data.

File: scripts/redshift/loader.py (hunk 1 above)
Comment (calvinnhieu) [reply scripts/redshift/loader.py]: Gotcha. Do you have a list of minimum supported versions for all schemas?

Comment (mckinsel) [reply scripts/redshift/loader.py]: Talked to @maniarathi about this on slack. The answer is that I do not have a list, and it's pretty tedious to create one. But, the minimum for `specimen` is probably the only relevant one.

**Pull request merged**